pkg/piecestore/rpc/client/client.go

```diff
@@ -166,3 +179,16 @@ func (client *Client) sign(msg []byte) (signature []byte, err error) {
 	// use c.pkey to sign msg
 	return cryptopasta.Sign(msg, client.prikey.(*ecdsa.PrivateKey))
 }
+
+// Route gets route
+func (client *Client) Route(ctx context.Context) (route pb.PieceStoreRoutesClient, err error) {
+	if client.route == nil && client.conn == nil {
```

Comment:
Perhaps just check `if client.route == nil`. It's valid for the `client.conn` to be nil if the PSClient was created with `NewCustomRoute`. The latter is used by unit tests which use mock route.